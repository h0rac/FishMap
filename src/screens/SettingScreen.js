import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { CheckBox, Divider } from 'react-native-elements';
import {
	StyleSheet,
	Text,
	View,
	TouchableOpacity,
	Switch,
  AsyncStorage
} from 'react-native';

import IconAwesome from 'react-native-vector-icons/FontAwesome';
import { connect } from 'react-redux';
import {
	emitWaypointReceive,
	setIntervalID,
} from '../actions/user';


class SettingScreen extends Component {


	constructor() {
		super();

		this.setEmit = this.setEmit.bind(this);

		this.state = {
			interval: 5000,
			intervalId: null,
			emitStatus: false,
			durationSet: false,
			color:'mintcream',
			pickerValue: 0
		};
	}

	static navigationOptions = ({ navigation }) => {

		const { params = {} } = navigation.state;

		return {
			headerStyle: {
				backgroundColor: '#2F95D6'
			},
			title: 'Settings',
			headerTintColor: 'white',
		};
	};
	static propTypes = {
		navigation: PropTypes.object
	};


	setEmit() {
		if (this.props.emitStatus) {
			this.props.dispatch(emitWaypointReceive(false));
			AsyncStorage.getItem('token').then(token => {
				if (token) {
					const intervalId = setInterval(() => this.props.socketIO.emit('onFishmarkUpdate', {
						token: JSON.parse(decodeURI(token)),
						receive: this.props.receive
					}), this.props.duration);
					this.props.dispatch(setIntervalID(intervalId));
				}
			});
		}
		else {
			clearInterval(this.props.timeoutID);
			this.props.dispatch(emitWaypointReceive(true));

		}
	}


	render() {

		return (
			<View style={styles.container}>
				<View style={styles.section}>
					<Text style={styles.sectionText}>GENERAL</Text>
				</View>
				<View style={styles.generalContainer}>
					<View style={styles.generalText}>
						<Text style={styles.enableText}>Enable waypoint updates</Text>
					</View>
					<View style={styles.enableSwitch}>
						<Switch
							key={'waypointSwitch'}
							value={!this.props.emitStatus}
							onValueChange={() => this.setEmit()}
						/>
					</View>
				</View>
				<Divider style={{ height:0.5, backgroundColor: 'gray'}} />

				<TouchableOpacity key='language' style={styles.generalContainer} onPress={()=>this.props.navigation.navigate('LanguageScreen')}>

					<View style={styles.generalText}>
						<Text style={styles.enableText}>Language settings</Text>
					</View>
					<View style={styles.enableSwitch}>
						<IconAwesome
							name={'angle-right'}
							size={28}
							color={'gray'}
						/>
					</View>
				</TouchableOpacity>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		backgroundColor: 'whitesmoke',
		flexDirection:'column'
	},

	generalContainer: {
		flex: 0.1,
		flexDirection:'row',
		backgroundColor:'mintcream',
		alignItems:'center',
		height:70
	},


	itemContainer: {
		flex: 0.1,
		flexDirection:'row',
		backgroundColor:'mintcream',
		alignItems:'center',
		paddingBottom:5,
		paddingTop:5
	},

	section: {
		backgroundColor: 'whitesmoke',
		flex:0.05,
		paddingLeft:20,
		paddingTop:30,
		flexDirection:'row',
		alignItems:'center'
	},

	enableSwitch: {
		flex: 1,
		flexDirection: 'row',
		justifyContent: 'flex-end',
		paddingRight:20

	},

	sectionText: {
		color:'gray'

	},

	generalText: {
		flex: 3,
		flexDirection: 'row',
		justifyContent: 'flex-start',
		paddingLeft:20
	},

	enableText: {
		paddingTop:5
	},

	title: {
		backgroundColor: 'black',
		color: '#fff',
		fontSize: 18,
		fontWeight: 'bold',
		padding: 10,
		textAlign: 'center'
	}
});


const mapStateToProps = state => {
	return {
		duration: state.user.duration,
		receive: state.user.receive,
		emitStatus: state.user.emitStatus,
		tempDuration: state.user.tempDuration,
		socketIO: state.user.socketIO,
		timeoutID: state.user.timeoutID
	};
};


export default connect(mapStateToProps)(SettingScreen);