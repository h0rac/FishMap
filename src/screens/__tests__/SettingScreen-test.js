import React from 'react';
import SettingScreen from '../SettingScreen';
import { AsyncStorage } from 'react-native';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import SocketIOClient from 'socket.io-client';
import mockedSocket from 'socket-io-mock';

configure({ adapter: new Adapter() });

const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('react-native-vector-icons/Ionicons', () => ({}));

jest.mock('react-native-vector-icons/FontAwesome', () => ({}));

jest.mock('AsyncStorage', () => {
	return {
		getItem: jest.fn((item) => {
			return new Promise((resolve, reject) => {
				resolve('"1234"');
			});
		})
	};
});


describe('Testing Main screen', () => {

	let tree;
	let navigate;
	let state;
	let props;
	let dispatch;

	const event = {
		nativeEvent: {
			coordinate: {
				latitude: 54.475408,
				longitude: 18.263086,
				latitudeDelta: 0.0922,
				longitudeDelta: 0.0421
			}
		}
	};

	beforeEach(() => {
		navigate = jest.fn();
		state = {
			user: { duration: 1000, receive: false }
		};
		const store = mockStore(state);
		dispatch = jest.fn();

		props = store;

		tree = shallow(
			<SettingScreen {...props} store={store} navigation={{ navigate }}
			/>, { lifecycleExperimental: true });
	});

	it('renders correctly', () => {
		expect(tree.dive()).toMatchSnapshot();
	});

	it('should render 1 Switch', () => {
		const SettingScreenCompononent = tree.dive();
		const switches = SettingScreenCompononent.find('Switch');
		expect(switches.length).toBe(1);
	});


	it('should start emit onFishmark update when switch is checked', () => {
		const SettingScreenCompononent = tree.dive();
		const switches = SettingScreenCompononent.find('Switch');
		const socket = new mockedSocket();
		SettingScreenCompononent.setProps({ emitStatus: true, socketIO: socket });
		switches.forEach(item => {
			if (item.getElement().key === 'waypointSwitch') {
				item.simulate('ValueChange');
				socket.socketClient.emit('onFishmarkUpdate', {
					token: JSON.parse(decodeURI('1234')),
					receive: true
				});
			}
		});
	});

	it('should stop emit onFishmark update when switch is unchecked', () => {
		const SettingScreenCompononent = tree.dive();
		const switches = SettingScreenCompononent.find('Switch');
		SettingScreenCompononent.setProps({ emitStatus: false });
		switches.forEach(item => {
			if (item.getElement().key === 'waypointSwitch') {
				item.simulate('ValueChange');
			}
		});
	});

	it('should move to Language screen when clicked Language setting', () => {
		const SettingScreenCompononent = tree.dive();
		const touchables = SettingScreenCompononent.find('TouchableOpacity');
		touchables.forEach(item => {
			if (item.getElement().key === 'language') {
				item.simulate('Press');
			}
		});
	});

});