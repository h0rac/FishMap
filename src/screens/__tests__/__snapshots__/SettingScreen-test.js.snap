// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main screen renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SettingScreen
    clearActions={[Function]}
    dispatch={[Function]}
    duration={1000}
    getActions={[Function]}
    getState={[Function]}
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    receive={false}
    replaceReducer={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      }
    }
    subscribe={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "whitesmoke",
              "flex": 0.05,
              "flexDirection": "row",
              "paddingLeft": 10,
              "paddingTop": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "gray",
              }
            }
          >
            GENERAL
          </Text>
        </Component>,
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            }
          }
        >
          <Component
            style={
              Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "paddingTop": 5,
                }
              }
            >
              Enable waypoint updates
            </Text>
          </Component>
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "paddingRight": 10,
              }
            }
          >
            <Switch
              disabled={false}
              onValueChange={[Function]}
              value={true}
            />
          </Component>
        </Component>,
        <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            }
          }
        >
          <Component
            style={
              Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "paddingTop": 5,
                }
              }
            >
              Language settings
            </Text>
          </Component>
          <Component
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "paddingRight": 10,
              }
            }
          >
            <UNDEFINED
              color="gray"
              name="angle-right"
              size={28}
            />
          </Component>
        </TouchableOpacity>,
      ],
      "style": Object {
        "backgroundColor": "whitesmoke",
        "flex": 1,
        "flexDirection": "column",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "gray",
              }
            }
          >
            GENERAL
          </Text>,
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "whitesmoke",
            "flex": 0.05,
            "flexDirection": "row",
            "paddingLeft": 10,
            "paddingTop": 30,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "GENERAL",
            "ellipsizeMode": "tail",
            "style": Object {
              "color": "gray",
            },
          },
          "ref": null,
          "rendered": "GENERAL",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingTop": 5,
                  }
                }
              >
                Enable waypoint updates
              </Text>
            </Component>,
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                }
              }
            >
              <Switch
                disabled={false}
                onValueChange={[Function]}
                value={true}
              />
            </Component>,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "mintcream",
            "flex": 0.1,
            "flexDirection": "row",
            "height": 70,
            "marginBottom": 2,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingTop": 5,
                  }
                }
              >
                Enable waypoint updates
              </Text>,
              "style": Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Enable waypoint updates",
                "ellipsizeMode": "tail",
                "style": Object {
                  "paddingTop": 5,
                },
              },
              "ref": null,
              "rendered": "Enable waypoint updates",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Switch
                disabled={false}
                onValueChange={[Function]}
                value={true}
              />,
              "style": Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "paddingRight": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": "waypointSwitch",
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "onValueChange": [Function],
                "value": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "language",
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": Array [
            <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingTop": 5,
                  }
                }
              >
                Language settings
              </Text>
            </Component>,
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                }
              }
            >
              <UNDEFINED
                color="gray"
                name="angle-right"
                size={28}
              />
            </Component>,
          ],
          "onPress": [Function],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "mintcream",
            "flex": 0.1,
            "flexDirection": "row",
            "height": 70,
            "marginBottom": 2,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingTop": 5,
                  }
                }
              >
                Language settings
              </Text>,
              "style": Object {
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "flex-start",
                "paddingLeft": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Language settings",
                "ellipsizeMode": "tail",
                "style": Object {
                  "paddingTop": 5,
                },
              },
              "ref": null,
              "rendered": "Language settings",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <UNDEFINED
                color="gray"
                name="angle-right"
                size={28}
              />,
              "style": Object {
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "flex-end",
                "paddingRight": 10,
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "color": "gray",
                "name": "angle-right",
                "size": 28,
              },
              "ref": null,
              "rendered": null,
              "type": Object {},
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "whitesmoke",
                "flex": 0.05,
                "flexDirection": "row",
                "paddingLeft": 10,
                "paddingTop": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "gray",
                }
              }
            >
              GENERAL
            </Text>
          </Component>,
          <Component
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "mintcream",
                "flex": 0.1,
                "flexDirection": "row",
                "height": 70,
                "marginBottom": 2,
              }
            }
          >
            <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingTop": 5,
                  }
                }
              >
                Enable waypoint updates
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                }
              }
            >
              <Switch
                disabled={false}
                onValueChange={[Function]}
                value={true}
              />
            </Component>
          </Component>,
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "mintcream",
                "flex": 0.1,
                "flexDirection": "row",
                "height": 70,
                "marginBottom": 2,
              }
            }
          >
            <Component
              style={
                Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "paddingTop": 5,
                  }
                }
              >
                Language settings
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                }
              }
            >
              <UNDEFINED
                color="gray"
                name="angle-right"
                size={28}
              />
            </Component>
          </TouchableOpacity>,
        ],
        "style": Object {
          "backgroundColor": "whitesmoke",
          "flex": 1,
          "flexDirection": "column",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "gray",
                }
              }
            >
              GENERAL
            </Text>,
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "whitesmoke",
              "flex": 0.05,
              "flexDirection": "row",
              "paddingLeft": 10,
              "paddingTop": 30,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "GENERAL",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "gray",
              },
            },
            "ref": null,
            "rendered": "GENERAL",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "flex": 3,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingLeft": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "paddingTop": 5,
                    }
                  }
                >
                  Enable waypoint updates
                </Text>
              </Component>,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "paddingRight": 10,
                  }
                }
              >
                <Switch
                  disabled={false}
                  onValueChange={[Function]}
                  value={true}
                />
              </Component>,
            ],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "paddingTop": 5,
                    }
                  }
                >
                  Enable waypoint updates
                </Text>,
                "style": Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Enable waypoint updates",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "paddingTop": 5,
                  },
                },
                "ref": null,
                "rendered": "Enable waypoint updates",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Switch
                  disabled={false}
                  onValueChange={[Function]}
                  value={true}
                />,
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": "waypointSwitch",
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "onValueChange": [Function],
                  "value": true,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "language",
          "nodeType": "class",
          "props": Object {
            "activeOpacity": 0.2,
            "children": Array [
              <Component
                style={
                  Object {
                    "flex": 3,
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                    "paddingLeft": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "paddingTop": 5,
                    }
                  }
                >
                  Language settings
                </Text>
              </Component>,
              <Component
                style={
                  Object {
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "paddingRight": 10,
                  }
                }
              >
                <UNDEFINED
                  color="gray"
                  name="angle-right"
                  size={28}
                />
              </Component>,
            ],
            "onPress": [Function],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "mintcream",
              "flex": 0.1,
              "flexDirection": "row",
              "height": 70,
              "marginBottom": 2,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "paddingTop": 5,
                    }
                  }
                >
                  Language settings
                </Text>,
                "style": Object {
                  "flex": 3,
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "paddingLeft": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Language settings",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "paddingTop": 5,
                  },
                },
                "ref": null,
                "rendered": "Language settings",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <UNDEFINED
                  color="gray"
                  name="angle-right"
                  size={28}
                />,
                "style": Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "flex-end",
                  "paddingRight": 10,
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "color": "gray",
                  "name": "angle-right",
                  "size": 28,
                },
                "ref": null,
                "rendered": null,
                "type": Object {},
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
    "lifecycleExperimental": true,
  },
}
`;
